pkgs/development/web/hoppscotch/bin.nix

```diff
@@ -0,0 +1,61 @@
+{ appimageTools
+, fetchurl
+, lib
+, makeDesktopItem
+}:
+
+let
+  pname = "hoppscotch";
+  version = "23.8.3";
+  name = "${pname}-${version}";
+
+  desktopItem = makeDesktopItem {
+    categories = [ "Development" ];
+    desktopName = "Hoppscotch";
+    exec = pname;
+    icon = pname;
+    name = pname;
+  };
+
+  icon = fetchurl {
+    url = "https://raw.githubusercontent.com/hoppscotch/hoppscotch/de725337d628d037e8597c02ef51be466bec3680/packages/hoppscotch-common/public/logo.svg";
+    hash = "sha256-Njbc+RTKSOziXo0H2Mv7RyNI5CLZNkJLUr/PatyrK9E=";
+  };
+in
+appimageTools.wrapType1 {
+  inherit name version;
+
+  src = fetchurl {
+    url = "https://github.com/hoppscotch/releases/releases/download/v${version}-1/Hoppscotch_linux_x64.AppImage";
+    hash = "sha256-YqvGYn3TrR4dvx3nKkGRfV7srTllDR4Q+dxttucklUM=";
+  };
+  extraPkgs = pkgs: [ pkgs.libthai ];
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+
+    mkdir -p $out/share/icons/hicolor/scalable/apps
+    cp ${icon} $out/share/icons/hicolor/scalable/apps/${pname}.svg
+
+    mkdir -p $out/share/applications
+    cp -r ${desktopItem} $out/share/applications/${pname}.desktop
+  '';
+
+  meta = with lib; {
+    description = "👽 Open-source API development ecosystem";
+    longDescription = ''
+      Hoppscotch is a lightweight, web-based API development suite. It was built
+      from the ground up with ease of use and accessibility in mind providing
+      all the functionality needed for API developers with minimalist,
+      unobtrusive UI.
+    '';
+    homepage = "https://hoppscotch.com";
+    downloadPage = "https://hoppscotch.com/downloads";
+    changelog = "https://hoppscotch.com/changelog";
+    license = licenses.mit;
+    maintainers = with maintainers; [ getpsyched ];
+    mainProgram = pname;
+    platforms = platforms.all;
```

Comment:
Isn't the `meta.platforms` only Linux as the derivation downloads a Linux AppImage?

Reply:
Oops, my bad. I incorrectly assumed the AppImage to be platform agnostic. Pushing a fix real quick.

Reply:
I think it might still be incorrect. Since the AppImage specifies the architecture too?